react-accordion/accordion/src/TopicCard.tsx

```diff
@@ -0,0 +1,35 @@
+type Topic = {
+  id: number;
+  title: string;
+  content: string;
+};
+
+type Props = {
+  topic: Topic;
+  active: boolean;
+  onSelected: (id: number) => void;
+  onClosed: () => void;
```

Comment:
This works well. However, it requires both the Accordion and the TopicCard to know that clicking in an open card should close it. If this logic were all in the Accordion, only the Accordion would need to know this behavior. In that case, you could just have a single "onClick" (or onSelect) event.

Either approach works, but keeping all the logic in one place makes it easier to add features, and reuse the TopicCard component.